dashboard/app/helpers/curriculum_helper.rb

```diff
@@ -18,4 +18,13 @@ def validate_key_format
     end
     return true
   end
+
+  # retrieves the matching UnitGroup or Script associated with a course version and offering
+  def find_matching_course_version(course_name)
```

Comment:
Ran into a problem of not being able to just look up course version directly when there are extra "-"s in the key, so I borrowed this bit from vocabularies. Does this make sense to use for ref guides as well?

Reply:
Yes, I think so. The problem with CourseVersion is it doesn't have a good unique key to use in URLs, which is how this logic came about.

Reply:
Thinking more about this -- I think this is fine for now but when we launch reference guides, we'll probably want to make sure this function isn't doing too many queries. This was previously used for a levelbuilder-facing page, so caching wasn't something we thought about. Definitely doesn't need to block this PR though